app/api/ada/lib/state-fetch/batchedFetcher.js

```diff
@@ -0,0 +1,268 @@
+// @flow
+
+import BigNumber from 'bignumber.js';
+import type {
+  AddressUtxoRequest, AddressUtxoResponse,
+  TxBodiesRequest, TxBodiesResponse,
+  UtxoSumRequest, UtxoSumResponse,
+  HistoryRequest, HistoryResponse,
+  SignedRequest, SignedResponse,
+  FilterUsedRequest, FilterUsedResponse,
+  AddressUtxoFunc,
+  FilterFunc,
+  HistoryFunc,
+  TxBodiesFunc,
+  UtxoSumFunc,
+} from './types';
+
+import type {
+  Transaction
+} from '../../adaTypes';
+
+import type { IFetcher } from './IFetcher';
+
+import _ from 'lodash';
+import {
+  CheckAdressesInUseApiError,
+  GetAllUTXOsForAddressesError,
+  GetTxsBodiesForUTXOsError,
+  GetUtxosSumsForAddressesApiError,
+  GetTxHistoryForAddressesApiError,
+} from '../../errors';
+import {
+  Logger,
+  stringifyError
+} from '../../../../utils/logging';
+
+import type { ConfigType } from '../../../../../config/config-types';
+import config from '../../../../config';
+
+declare var CONFIG: ConfigType;
+const addressesLimit = CONFIG.app.addressRequestSize;
+const transactionsLimit: number = config.wallets.TRANSACTION_REQUEST_SIZE;
+
+/**
+ * Makes calls to Yoroi backend service
+ * https://github.com/Emurgo/yoroi-backend-service/
+ */
+export class BatchedFetcher implements IFetcher {
+
+  baseFetcher: IFetcher;
+
+  constructor(baseFetcher: IFetcher) {
+    this.baseFetcher = baseFetcher;
+  }
+
+  getUTXOsForAddresses = (body: AddressUtxoRequest): Promise<AddressUtxoResponse> => (
+    batchUTXOsForAddresses(this.baseFetcher.getUTXOsForAddresses)(body)
+  )
+
+  getTxsBodiesForUTXOs = (body: TxBodiesRequest): Promise<TxBodiesResponse> => (
+    batchTxsBodiesForInputs(this.baseFetcher.getTxsBodiesForUTXOs)(body)
+  )
+
+  getUTXOsSumsForAddresses = (body: UtxoSumRequest): Promise<UtxoSumResponse> => (
+    batchGetUTXOsSumsForAddresses(this.baseFetcher.getUTXOsSumsForAddresses)(body)
+  )
+
+  getTransactionsHistoryForAddresses = (body: HistoryRequest): Promise<HistoryResponse> => (
+    batchGetTransactionsHistoryForAddresses(
+      this.baseFetcher.getTransactionsHistoryForAddresses
+    )(body)
+  )
+
+  sendTx = (body: SignedRequest): Promise<SignedResponse> => (
+    // We don't batch transaction sending (it's just a single requeset)
+    // TODO: Should we support batching a list of transactions?
+    this.baseFetcher.sendTx(body)
```

Comment:
There once was an idea that we can eventually create a "tx constructor" in Yoroi, that would allow to define multiple targets and amounts first and then enter spending password once and create/send multiple transactions. There still will be a good discussion on whether we need a functionality like this, over an option to specify multiple outputs for a single transaction, but that, imo, is the only good potential use for batched tx sending, atm